src/App/Fossa/ManualDeps.hs

```diff
@@ -148,9 +149,16 @@ toSourceUnit root depsFile manualDeps@ManualDependencies{..} maybeApiOpts vendor
     -- Don't do anything if there are no vendored deps.
     (_, Nothing) -> pure []
 
+  -- Some manual deps, such as remote dependencies in source unit cannot be
+  -- validated without endpoint interactions.
+  rdeps <- case maybeApiOpts of
+    Just apiOpts -> runFossaApiClient apiOpts $ do
+      traverse (\r -> validateRemoteDep r =<< getOrganization) remoteDependencies
```

Comment:
🤔 I'm actually not sure about this, but does this result in `getOrganization` being called for every element in the list during `traverse`? If it wasn't monadic I think the answer would be no, but since it is I think it does?

If so please move the `getOrganization` outside of `traverse`.